service/controller/v13/resource/instance/create.go

```diff
@@ -33,754 +39,42 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	if err != nil {
 		return microerror.Mask(err)
 	}
-	deploymentsClient, err := r.getDeploymentsClient(ctx)
```

Comment:
The diff for this file is really bad. All the removed code was moved into other files and only the driving code for state machine is now in `EnsureCreated()` function. Here it is after changes:
```go
func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
        customObject, err := key.ToCustomObject(obj)
        if err != nil {
                return microerror.Mask(err)
        }

        var newState state.State
        var currentState state.State
        {
                s, err := r.getResourceStatus(customObject, Stage)
                if err != nil {
                        return microerror.Mask(err)
                }
                currentState = state.State(s)

                if currentState == "" {
                        // DeploymentUninitialized is the initial state for instance resource.
                        newState = DeploymentUninitialized
                        r.logger.LogCtx(ctx, "level", "debug", "message", "no current state present")
                } else {
                        r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("current state: %s", currentState))
                        newState, err = r.stateMachine.Execute(ctx, obj, currentState)
                        if err != nil {
                                return microerror.Mask(err)
                        }
                }
        }

        if newState != currentState {
                r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("new state: %s", newState))
                r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("setting resource status to '%s/%s'", Stage, newState))
                err = r.setResourceStatus(customObject, Stage, string(newState))
                if err != nil {
                        return microerror.Mask(err)
                }
                r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("set resource status to '%s/%s'", Stage, newState))
                r.logger.LogCtx(ctx, "level", "debug", "message", "canceling reconciliation")
                reconciliationcanceledcontext.SetCanceled(ctx)
        } else {
                r.logger.LogCtx(ctx, "level", "debug", "message", "no state change")
        }

        return nil
}
```